### DISCUSSION TITLE
Renovatebot does not find the files in a different folder

### DISCUSSION THREAD

**Discussion by klaasnicolaas** [renovatebot/renovate#22529] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

What I want to achieve is that renovate bot finds my github actions files in another folder and updates the versions from now on.

https://github.com/klaasnicolaas/pypackage-template/blob/main/.github/renovate.json

In my template repository I have github actions files in 2 different places, `.github/workflows` and `template/.github/workflows`, only the first path can find it but not the ones in the template folder.

### What you tried so far.

- Adjusted the default file matching for the managers, had no effect.
- Applied `matchFiles` and `matchPaths` in the packageRules with the correct link to the location where the files should be, had no effect.

Also attached is a piece of the debug log from the developers mend dashboard which shows that it always finds 5 files coming from the .github folder in the main folder, but not those in the template folder. The commit history also shows what I've tried.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Matched 5 file(s) for manager github-actions: .github/workflows/labels.yaml, .github/workflows/lock.yaml, .github/workflows/pr-labels.yaml, .github/workflows/release-drafter.yaml, .github/workflows/stale.yaml
```

</details>

**Comment by rarkins** [comment]
How have you tested your regex? e.g. in regex101.com?

The string `\\template` looks wrong, for example

**Reply by rarkins** [reply, marked as answer]
The default fileMatch you have copy/pasted from is:

```
  fileMatch: [
    '^(workflow-templates|\\.github/workflows)/[^/]+\\.ya?ml$',
    '(^|/)action\\.ya?ml$',
  ],
```

You could do this:

```
  fileMatch: [
    '^(workflow-templates|(template/)?\\.github/workflows)/[^/]+\\.ya?ml$',
    '(^|/)action\\.ya?ml$',
  ],
```

Test: https://regex101.com/r/gAdaNw/1

**Reply by klaasnicolaas** [reply]
The regex was not tested because I don't have much experience with it, but I will definitely keep that website in mind for the future. I modified it and Renovate bot now finds the files in the template folder, so thanks a lot for the help!

I now also need to apply the same for a [pyproject.toml](https://github.com/klaasnicolaas/pypackage-template/blob/main/template/pyproject.toml.j2) file in the template folder, the only thing is that this is also a jinja file and I was wondering if renovate bot can also handle that so that the dependencies still could be updated?

**Reply by rarkins** [reply]
It's preferred to keep to 1 topic per Discussion, so as to benefit and not confused future readers. I will answer this but then hide these comments as off-topic, so please start a new discussion if in doubt. The answer is no: if Renovate is using a YAML parser for a manager, and the templating means it's invalid YAML, then it won't be detected/parsed.
